### PR TITLE
I03 basic devices

### PR DESCRIPTION
**Pull request by npat113**
Fixes # (issue)
basic starting devices to allow dev across branches

### To test:
everything

### CODE REVIEW

File: src/artemis/devices/I03Smargon.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsMotor, EpicsSignal
+
+"""
+Real motors added to allow stops following pin load (e.g. real_x1.stop() )
+Stub offsets are calibration values that are required to move between calibration pin position and spine pins. These are set in EPCICS and applied via the proc.
+"""
+
+
+class I03Smargon(Device):
```
Comment (DominicOram) [inline src/artemis/devices/I03Smargon.py:10]: Should: There is already an I03 smargon in `devices/motors.py`. Could you add any additionally required PVs there? Or if you think it makes sense to have it in a separate smargon file we can do that too. Just don't want two copies

File: src/artemis/move_det_robot_load.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import bluesky.plan_stubs as bps
```
Comment (DominicOram) [inline src/artemis/move_det_robot_load.py:1]: Should: I think this is part of https://github.com/DiamondLightSource/python-artemis/issues/335 right? Can we do it in a separate PR for that rather than here?

**Review by DominicOram — CHANGES_REQUESTED** [review]
Couple of minor suggestions. You also have quite a few empty files:
* DCM
* beamstop
* cryostream
* fluorescence_detector_motion
* lower_gonio_stages
* qbpm1
* robot
* scintillator

I would rather not have empty files in the codebase as it leads to confusion. If you have some code for them now feel free to add to this PR, if not I suggest we remove from this PR and add when required later.

File: src/artemis/devices/I03Smargon.py (hunk 1 above)
Comment (npat113) [reply src/artemis/devices/I03Smargon.py:10]: I think there is a discussion to be had on devices and how granular you want them in terms of file structure. Motors.py is either going to become very large, or should be split to individual devices. Should something similar to the EPICS grouping be used?

File: src/artemis/move_det_robot_load.py (hunk 2 above)
Comment (npat113) [reply src/artemis/move_det_robot_load.py:1]: yes, not sure how that sneaked in there

File: src/artemis/devices/I03Smargon.py (hunk 1 above)
Comment (DominicOram) [reply src/artemis/devices/I03Smargon.py:10]: Yh, I agree that motors was probably too large a category. It made sense when the usecase was just grid scan but now we're looking wider afield it doesn't make as much sense. Happy to move it to here. In which case can you merge the one in `motors.py` and change all references? I can do this if you'd rather.

File: src/artemis/devices/CTAB.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsMotor, EpicsSignalRO
+
+
+class CTAB(Device):
+    """Basic collimantion table (CTAB) device for motion plus the motion disable signal when laser curtain triggered and hutch not locked"""
+
+    inboard_y: EpicsMotor = Cpt(EpicsMotor, "-MO-TABLE-01:INBOARDY")
```
Comment (DominicOram) [inline src/artemis/devices/CTAB.py:8]: Nit: Do you know if there's anywhere this co-ord system is documented? I can write some down if not

File: src/artemis/devices/DCM.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsMotor, EpicsSignalRO
+
+
+class DCM(Device):
+
+    # upstream_x: EpicsMotor = Cpt(EpicsMotor, '-MO-DET-01:UPSTREAMX')
+
```
Comment (DominicOram) [inline src/artemis/devices/DCM.py:8]: Nit: Is this used here? We should remove it if not

File: src/artemis/devices/DCM.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    bragg: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:BRAGG")
+    roll: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:ROLL")
+    offset: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:OFFSET")
+    perp: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:PERP")
+    energy: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:ENERGY")
+    pitch: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:PITCH")
+    wavelength: EpicsMotor = Cpt(EpicsMotor, "-MO-DCM-01:WAVELENGTH")
```
Comment (DominicOram) [inline src/artemis/devices/DCM.py:15]: Nit: Do you know of any documentation for what each of these do? If not can we write some comments? The ones in particular I'm not so sure about:
* offset
* perp

File: src/artemis/devices/I03Smargon.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    y: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:Y")
+    z: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:Z")
+    chi: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:CHI")
+    phi: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:PHI")
+    omega: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:OMEGA")
+
+    stub_offset_set: EpicsSignal = Cpt(EpicsSignal, "-MO-SGON-01:SET_STUBS_TO_RL.PROC")
+
+    real_x1: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:MOTOR_3")
+    real_x2: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:MOTOR_4")
```
Comment (DominicOram) [inline src/artemis/devices/I03Smargon.py:21]: Nit: Could you put in a comment about what the difference is between `x1` and `x2`?

File: src/artemis/devices/I03Smargon.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    real_y: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:MOTOR_1")
+    real_z: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:MOTOR_2")
+    real_phi: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:MOTOR_5")
+    real_chi: EpicsMotor = Cpt(EpicsMotor, "-MO-SGON-01:MOTOR_6")
```
Comment (DominicOram) [inline src/artemis/devices/I03Smargon.py:25]: Nit: Why do we need the real motors here? I would have thought we only ever really want to use the virtual axes?

File: src/artemis/devices/fluorescence_detector_motion.py [hunk 8]
```diff
@@ -0,0 +1,7 @@
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal
+
+
+class FluorescenceDetector(Device):
+
+    pos: EpicsSignal = Cpt(EpicsSignal, "-EA-FLU-01:CTRL")
```
Comment (DominicOram) [inline src/artemis/devices/fluorescence_detector_motion.py:7]: Nit: Pos implies a position, which I would assume is a motor but I'm guessing this is just an IN/OUT? In which case could you do something like the backlight where you define:
```
class Backlight(Device):
    OUT = 0
    IN = 1

    pos: EpicsSignal = Component(EpicsSignal, "-EA-BL-01:CTRL")
```

File: src/artemis/move_det_robot_load.py (hunk 2 above)
Comment (DominicOram) [reply src/artemis/move_det_robot_load.py:1]: You can remove it by doing `git rm --cached src/artemis/move_det_robot_load.py`. This will take it out of git but keep it locally if you're worried about losing it.

**Review by DominicOram — CHANGES_REQUESTED** [review]
Thank you, mostly just requests for more documentation. Some of this may be my own lack of understanding but we should try and make it accessible to someone relatively new to DLS.

File: src/artemis/devices/DCM.py (hunk 4 above)
Comment (npat113) [reply src/artemis/devices/DCM.py:8]: no, that comment has sneaked in from elsewhere. Will remove

**Review by DominicOram — APPROVED** [review]
Great, thank you. I made some minor edits with moving the location of comments. If you put the comments in the right place then tooltips on IDEs will pick them up. So if you have a line that says `mv(smargon.stub_offset_set, 10)` with the comment located where it now is you can hover over `stub_offset_set` to get a description of what it is.

File: src/artemis/devices/I03Smargon.py (hunk 1 above)
Comment (DominicOram) [reply src/artemis/devices/I03Smargon.py:5]: I will merge this PR and fix this in a subsequent issue
